harness/determined/_execution.py

```diff
@@ -16,6 +16,18 @@ def _get_gpus() -> Tuple[bool, List[str], List[int]]:
     return use_gpu, gpu_uuids, gpu_ids
 
 
+def _execute_user_function(func: Callable, *args: Any, **kwargs: Any) -> Any:
```

Comment:
Can we make this a context manager, so it would look like:

```python
with puke_on_sys_exit():
     # code that may call user code here
```

Reply:
I think it's overkill to use a context manager here. `try..except` should be just enough.

Reply:
A context manager is just a convenient mechanism for a try/except anyway:

```python
@contextlib.contextmanager
def puke_on_system_exit() -> Generator:
    try:
        yield
    except SystemExit as e:
        ...
```

The context manager version does not affect mypy typing of the functions and it is easier to read and understand.

Reply:
done